src/Compilers/CSharp/Test/Symbol/Symbols/DefaultInterfaceImplementationTests.cs

```diff
@@ -5086,5 +5086,578 @@ class Test1 : I1
+        this.M3();
+        this.P3 = this.P3;
+        this.E3 += null;
+        this.E3 -= null;
+    }
+
+    int P2
+    {
+        get
+        {
+            System.Console.WriteLine(""I2.get_P2"");
+            System.Console.WriteLine(this.GetHashCode());
+            this.M1();
+            this.P1 = this.P1;
+            this.E1 += null;
+            this.E1 -= null;
+            this.M3();
+            this.P3 = this.P3;
+            this.E3 += null;
+            this.E3 -= null;
+            return 0;
+        }
+        set
+        {
+            System.Console.WriteLine(""I2.set_P2"");
+            System.Console.WriteLine(this.GetHashCode());
+            this.M1();
+            this.P1 = this.P1;
+            this.E1 += null;
+            this.E1 -= null;
+            this.M3();
+            this.P3 = this.P3;
+            this.E3 += null;
+            this.E3 -= null;
+        }
+    }
+
+    event System.Action E2
+    {
+        add
+        {
+            System.Console.WriteLine(""I2.add_E2"");
+            System.Console.WriteLine(this.GetHashCode());
+            this.M1();
+            this.P1 = this.P1;
+            this.E1 += null;
+            this.E1 -= null;
+            this.M3();
+            this.P3 = this.P3;
+            this.E3 += null;
+            this.E3 -= null;
+        }
+        remove
+        {
+            System.Console.WriteLine(""I2.remove_E2"");
+            System.Console.WriteLine(this.GetHashCode());
+            this.M1();
+            this.P1 = this.P1;
+            this.E1 += null;
+            this.E1 -= null;
+            this.M3();
+            this.P3 = this.P3;
+            this.E3 += null;
+            this.E3 -= null;
+        }
+    }
+
+    void M3() 
+    {
+        System.Console.WriteLine(""I2.M3"");
+    }
+
+    int P3
+    {
+        get
+        {
+            System.Console.WriteLine(""I2.get_P3"");
+            return 0;
+        }
+        set => System.Console.WriteLine(""I2.set_P3"");
+    }
+
+    event System.Action E3
+    {
+        add => System.Console.WriteLine(""I2.add_E3"");
+        remove => System.Console.WriteLine(""I2.remove_E3"");
+    }
+}
+
+
+class Test1 : I2
+{
+    static void Main()
+    {
+        I2 x = new Test1();
+        x.M2();
+        x.P2 = x.P2;
+        x.E2 += null;
+        x.E2 -= null;
+    }
+
+    public override int GetHashCode()
+    {
+        return 123;
+    }
+}
+";
+            var compilation1 = CreateCompilationWithMscorlib(source1, options: TestOptions.DebugDll,
+                                                             parseOptions: TestOptions.Regular.WithLanguageVersion(LanguageVersion.Latest));
+            Assert.True(compilation1.Assembly.RuntimeSupportsDefaultInterfaceImplementation);
+
+            CompileAndVerify(compilation1, verify: false);
+
+/* Expected output
+I2.M2
+123
+I1.M1
+I1.get_P1
+I1.set_P1
+I1.add_E1
+I1.remove_E1
+I2.M3
+I2.get_P3
+I2.set_P3
+I2.add_E3
+I2.remove_E3
+I2.get_P2
```

Comment:
Consider some additional formatting to improve readability. Maybe add "## " on the "headers".